tests/components/switchbot_cloud/test_vacuum.py

```diff
@@ -0,0 +1,440 @@
+"""Test for the switchbot_cloud vacuum."""
+
+from unittest.mock import patch
+
+from switchbot_api import Device
+
+from homeassistant.components.switchbot_cloud import SwitchBotAPI
+from homeassistant.components.vacuum import DOMAIN as VACUUM_DOMAIN, VacuumActivity
+from homeassistant.const import ATTR_ENTITY_ID, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+
+from . import configure_integration
+
+
+async def test_coordinator_data_is_none(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """Test coordinator data is none."""
+    mock_list_devices.return_value = [
+        Device(
+            version="V1.0",
+            deviceId="vacuum-id-1",
+            deviceName="vacuum-1",
+            deviceType="K10+",
+            hubDeviceId="test-hub-id",
+        ),
+    ]
+    mock_get_status.side_effect = [
+        None,
+    ]
+    await configure_integration(hass)
+    entity_id = "vacuum.vacuum_1"
+    state = hass.states.get(entity_id)
+
+    assert state.state == STATE_UNKNOWN
+
+
+async def test_k10_plus_set_fan_speed(
+    hass: HomeAssistant, mock_list_devices, mock_get_status
+) -> None:
+    """Test K10 plus set fan speed."""
+
+    mock_list_devices.side_effect = [
+        [
+            Device(
+                version="V1.0",
+                deviceId="vacuum-id-1",
+                deviceName="vacuum-1",
+                deviceType="K10+",
+                hubDeviceId="test-hub-id",
+            )
+        ]
+    ]
+    mock_get_status.side_effect = [
+        {
+            "deviceType": "K10+",
+            "workingStatus": "Cleaning",
+            "battery": 50,
+            "onlineStatus": "online",
+        },
+    ]
+
+    await configure_integration(hass)
+    entity_id = "vacuum.vacuum_1"
+    with patch.object(SwitchBotAPI, "send_command") as mock_send_command:
+        await hass.services.async_call(
+            VACUUM_DOMAIN,
+            "set_fan_speed",
+            {ATTR_ENTITY_ID: entity_id, "fan_speed": "quiet"},
+            blocking=True,
+        )
+        mock_send_command.assert_called()
```

Comment:
Since we now have different implementations of vacuums, would it make sense to actually test that we're correctly sending the right command?